app/routes/groups_.$groupId.tsx

```diff
@@ -7,14 +7,16 @@ import { LinkButton } from '~/components/ui/forms';
 import { Image } from '~/components/ui/images';
 import { eventsDataPatcher } from '~/modules/events/event';
 
-export const loader: LoaderFunction = async ({ params }) => {
-  const group = await db.group.findFirstOrThrow({ where: { id: params.groupId } });
-  const user_groups = await db.userGroup.findMany({ where: { groupId: group.id } });
-  return json({ group, user_groups });
-};
+export async function loader({ params }: LoaderFunctionArgs) {
+  const group = await db.group.findFirstOrThrow({
+    where: { id: params.groupId },
+    include: { events: true, user_groups: true },
+  });
+  return json({ group });
+}
 
 export default function GroupRoute() {
-  const { group, user_groups } = useLoaderData<typeof loader>();
+  const { group } = useLoaderData<typeof loader>();
```

Comment:
We may be able to optimize the amount of data sent to the client by updating our db query or manipulate our data before sending to the client.

Reply:
optimized a bit 👍